mythril/laser/ethereum/state.py

```diff
@@ -1,16 +1,62 @@
-from z3 import BitVec, BitVecVal, Solver, ExprRef, sat
+from z3 import BitVec, BitVecVal, BitVecRef, BitVecNumRef, BitVecSort, Solver, ExprRef, Concat, sat, simplify, Array
+from z3.z3types import Z3Exception
 from mythril.disassembler.disassembly import Disassembly
 from copy import copy, deepcopy
 from enum import Enum
 from random import randint
 
 from mythril.laser.ethereum.evm_exceptions import StackOverflowException, StackUnderflowException
 
-
 class CalldataType(Enum):
     CONCRETE = 1
     SYMBOLIC = 2
 
+class Calldata:
```

Comment:
shouldn't __len__() also be implemented for this class as calldatasize opcode uses it?

Reply:
Why was this resolved?

Reply:
He is storing calldatasize as a variable and is accessing it in that opcode, but now i am thinking it might be cleaner to have \_\_len\_\_() over having to use the class variables, what do you think?

Reply:
He added a variable calldatasize which he used in that opcode, now it makes me think that using \_\_len\_\_() might be cleaner as we don't need to remember that such a variable exists in the class, what do you think?

Reply:
That does sound cleaner. @norhh I'll implement it that way.

Reply:
@norhh Just kidding. I believe that in Python, `__len__` must return an integer, and I need to return a BitVec sometimes.